extensions/bitwise.js

```diff
@@ -1,113 +1,259 @@
+            opcode: 'bitwiseLogicalRightShift',
+            blockType: Scratch.BlockType.REPORTER,
+            text: '[LEFT] ⇶ [RIGHT]',
+            arguments: {
+              LEFT: {
+                type: Scratch.ArgumentType.NUMBER,
+                defaultValue: '\n'
+              },
+              RIGHT: {
+                type: Scratch.ArgumentType.NUMBER,
+                defaultValue: '\n'
+              }
+            }
+          },
+         {
+            opcode: 'bitwiseCircularRightShift',
             blockType: Scratch.BlockType.REPORTER,
-            text: '[LEFT] >> [RIGHT]',
+            text: '[LEFT] ↻ [RIGHT]',
             arguments: {
               LEFT: {
                 type: Scratch.ArgumentType.NUMBER,
-                defaultValue: '2'
+                defaultValue: '\n'
               },
               RIGHT: {
                 type: Scratch.ArgumentType.NUMBER,
-                defaultValue: '1'
+                defaultValue: '\n'
               }
             }
           },
+          {
+            opcode: 'bitwiseCircularLeftShift',
+            blockType: Scratch.BlockType.REPORTER,
+            text: '[LEFT] ↺ [RIGHT]',
+            arguments: {
+              LEFT: {
+                type: Scratch.ArgumentType.NUMBER,
+                defaultValue: '\n'
+              },
+              RIGHT: {
+                type: Scratch.ArgumentType.NUMBER,
+                defaultValue: '\n'
+              }
+            }
+          },
+          '---',
           {
             opcode: 'bitwiseAnd',
             blockType: Scratch.BlockType.REPORTER,
-            text: '[LEFT] & [RIGHT]',
+            text: '[LEFT] and [RIGHT]',
             arguments: {
               LEFT: {
                 type: Scratch.ArgumentType.NUMBER,
-                defaultValue: '3'
+                defaultValue: '\n'
               },
               RIGHT: {
                 type: Scratch.ArgumentType.NUMBER,
-                defaultValue: '1'
+                defaultValue: '\n'
               }
             }
           },
           {
             opcode: 'bitwiseOr',
             blockType: Scratch.BlockType.REPORTER,
-            text: '[LEFT] | [RIGHT]',
+            text: '[LEFT] or [RIGHT]',
             arguments: {
               LEFT: {
                 type: Scratch.ArgumentType.NUMBER,
-                defaultValue: '1'
+                defaultValue: '\n'
               },
               RIGHT: {
                 type: Scratch.ArgumentType.NUMBER,
-                defaultValue: '0'
+                defaultValue: '\n'
               }
             }
           },
           {
             opcode: 'bitwiseXor',
             blockType: Scratch.BlockType.REPORTER,
-            text: '[LEFT] ^ [RIGHT]',
+            text: '[LEFT] xor [RIGHT]',
             arguments: {
               LEFT: {
                 type: Scratch.ArgumentType.NUMBER,
-                defaultValue: '1'
+                defaultValue: '\n'
```

Comment:
empty strings are supposed to work here now, is it still broken?

Reply:
Oh, sorry, just out of habit